lang/java/avro/src/main/java/org/apache/avro/file/CodecFactory.java

```diff
@@ -66,7 +66,7 @@ public static CodecFactory xzCodec(int compressionLevel) {
   /** Snappy codec. */
   public static CodecFactory snappyCodec() {
     try {
-      return new SnappyCodec.Option();
+      return SnappyCodec.OPTION;
     } catch (Throwable t) {
       LOG.debug("Snappy was not available", t);
       return null;
```

Comment:
In the new situation, the `SnappyCodec.OPTION` has already been created, so the try-catch block is now misplaced. Worse, Avro will fail to initialise when Snappy is not available.